test/rules/non_constant_identifier_names_test.dart

```diff
@@ -21,32 +21,32 @@ class NonConstantIdentifierNamesRecordsTest extends LintRuleTest {
   @override
   String get lintRule => 'non_constant_identifier_names';
 
-  @FailingTest(issue: 'https://github.com/dart-lang/linter/issues/3621')
   test_recordFields() async {
     await assertDiagnostics(r'''
 var a = (x: 1);
-var b = (X: 1);
+var b = (XXX: 1);
```

Comment:
Do we need long names? (Just a different style than we follow in the other packages, where we keep the test cases as short as possible.)

Reply:
Just greater than 1.  I'll update.